### PR TITLE
Removed unused ssl_dane struct declaration.

### PR DESCRIPTION
**Pull request by davxy**
The actually used structure is named ssl_dane_st.

CLA: trivial

### CODE REVIEW

**Comment by davxy** [comment]
I'm study the project code starting from the ground up.
I like clean-code, so its likely I'll find something else here and there :-)

File: include/openssl/types.h [hunk 1]
```diff
@@ -227,4 +227,5 @@ typedef unsigned long ossl_uintmax_t;
 #ifdef  __cplusplus
 }
 #endif
-#endif                          /* def OPENSSL_TYPES_H */
+
+#endif /* OPENSSL_TYPES_H */
```
Comment (mspncp) [inline include/openssl/types.h]: Note: unfortunately, this unnecessary unrelated change caused a conflict when cherry-picking to 1.1.1. Please raise a separate pull request for 1.1.1, if you want this patch to be backported.

**Comment by mspncp** [comment]
Merged to master in 96fc458a9cbeb8e7b30d789ea5919c5dcb9b980e, thank you!
